src/main/java/com/cloudbees/jenkins/support/util/WordReplacer.java

```diff
@@ -0,0 +1,168 @@
+        }
+
+        for(int i = 0; i < words.length; i++) {
+            replaceWordIgnoreCase(input, words[i], replaces[i]);
+        }
+    }
+
+    /**
+     * Replace all occurrences of word by replace in the input. The replacement happens only if the word to replace is
+     * separated from others words, that is, it's not part of a word. The implementation is that the previous and next
+     * characters of the word must be a {@link Character#isLetterOrDigit(char)} char.
+     * @param input text where the replacements take place
+     * @param word the word to replace
+     * @param replace the new text to use
+     */
+    public String replaceWord(String input, String word, String replace) {
+        StringBuilder sb = new StringBuilder(input);
+        replaceWord(sb, word, replace);
+        return sb.toString();
+    }
+
+    /**
+     * Replace all occurrences of word by replace in the input. The replacement happens only if the word to replace is
+     * separated from others words, that is, it's not part of a word. The implementation is that the previous and next
+     * characters of the word must be a {@link Character#isLetterOrDigit(char)} char.
+     * @param input text where the replacements take place
+     * @param word the word to replace
+     * @param replace the new text to use
+     */
+    public String replaceWordIgnoreCase(String input, String word, String replace) {
+        StringBuilder sb = new StringBuilder(input);
+        replaceWordIgnoreCase(sb, word, replace);
+        return sb.toString();
+    }
+
+    /**
+     * Replace all occurrences of word by replace in the input. The replacement happens only if the word to replace is
+     * separated from others words, that is, it's not part of a word. The implementation is that the previous and next
+     * characters of the word must be a {@link Character#isLetterOrDigit(char)} char.
+     * @param input text where the replacements take place
+     * @param word the word to replace
+     * @param replace the new text to use
+     */
+    public static void replaceWord(StringBuilder input, String word, String replace) {
+        replaceWord(input, word, replace, false);
+    }
+
+
+    public static void replaceWordIgnoreCase(StringBuilder input, String word, String replace ) {
+        replaceWord(input, word, replace, true);
+    }
+
+    // Main method
+    private static void replaceWord(StringBuilder input, String word, String replace, boolean ignoreCase) {
+        // The string where we look for
+        StringBuilder workInput;
+        // The string to find (lower case if ignoreCase)
+        String workWord = word;
+
+        // Use the lowercase versions to compare, but replace in the original input
+        if (ignoreCase) {
+            workInput = new StringBuilder(input.toString().toLowerCase(Locale.ENGLISH));
+            workWord = word.toLowerCase(Locale.ENGLISH);
```

Comment:
Replacing in some languages is not so trivial, the string can change its length and we are looking backward and forward for the word separator characters.